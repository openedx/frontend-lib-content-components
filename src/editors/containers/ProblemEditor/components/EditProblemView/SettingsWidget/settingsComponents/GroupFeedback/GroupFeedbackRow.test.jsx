import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import { formatMessage } from '../../../../../../../../testUtils';
import { GroupFeedbackRow } from './GroupFeedbackRow';

jest.mock('@edx/paragon', () => ({
  ...jest.requireActual('@edx/paragon'),
  Row: 'Row',
  IconButton: 'IconButton',
  Icon: 'Icon',
  Form: {
    CheckboxSet: 'Form.CheckboxSet',
    Checkbox: 'Form.CheckboxSet',
    Control: 'Form.Control',
  },
  ActionRow: 'ActionRow',
}));
jest.mock('@edx/paragon/icons', () => ({
  ...jest.requireActual('@edx/paragon/icons'),
  DeleteOutline: 'DeleteOutline',
}));

describe('GroupFeedbackRow', () => {
  const props = {
    value: { answers: ['A', 'C'], feedback: 'sOmE FeEDBACK' },
    answers: ['A', 'B', 'C', 'D'],
    handleAnswersSelectedChange: jest.fn().mockName('handleAnswersSelectedChange'),
    handleFeedbackChange: jest.fn().mockName('handleFeedbackChange'),
    handleDelete: jest.fn().mockName('handleDelete'),
    intl: { formatMessage },
  };

  describe('snapshot', () => {
    test('snapshot: renders hints row', () => {
      expect(shallow(<GroupFeedbackRow {...props} />).snapshot).toMatchSnapshot();
    });
  });
});
