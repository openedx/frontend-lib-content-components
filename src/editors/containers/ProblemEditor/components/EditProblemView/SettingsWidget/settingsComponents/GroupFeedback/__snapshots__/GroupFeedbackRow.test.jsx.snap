// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupFeedbackRow snapshot snapshot: renders hints row 1`] = `
<div
  className="mb-4"
>
  <ActionRow
    className="mb-2"
  >
    <Form.Control
      onChange={[MockFunction handleFeedbackChange]}
      value="sOmE FeEDBACK"
    />
    <div
      className="d-flex flex-row flex-nowrap"
    >
      <IconButton
        alt="Delete answer"
        iconAs="Icon"
        onClick={[MockFunction handleDelete]}
        src="DeleteOutline"
        variant="primary"
      />
    </div>
  </ActionRow>
  <Form.CheckboxSet
    onChange={[MockFunction handleAnswersSelectedChange]}
    value={
      Array [
        "A",
        "C",
      ]
    }
  >
    <Row
      className="mx-0"
    >
      <Form.CheckboxSet
        checked={-1}
        className="mr-4 mt-1"
        isValid={false}
      >
        <div
          className="x-small"
        />
      </Form.CheckboxSet>
      <Form.CheckboxSet
        checked={-1}
        className="mr-4 mt-1"
        isValid={false}
      >
        <div
          className="x-small"
        />
      </Form.CheckboxSet>
      <Form.CheckboxSet
        checked={-1}
        className="mr-4 mt-1"
        isValid={false}
      >
        <div
          className="x-small"
        />
      </Form.CheckboxSet>
      <Form.CheckboxSet
        checked={-1}
        className="mr-4 mt-1"
        isValid={false}
      >
        <div
          className="x-small"
        />
      </Form.CheckboxSet>
    </Row>
  </Form.CheckboxSet>
</div>
`;
